### PR TITLE
Detect image orientation

### PR DESCRIPTION
**Pull request by zcei**
Hej there,

we have been successfully using your library for the last months for detecting image dimensions. So thanks for the work that has been put in!

With the rise of macOS Mojave, we got some complaints about "swapped" width & height.

I started investigating and found out, that it now supports rotating images in Quick View that in turn sets the `Orientation` EXIF tag (compared to e.g. opening in Preview and rotating, which renders a new image).

This PR aims for a first implementation of detecting the image dimensions.
It is based on [this research by TsuruZoh Tachibanaya](https://www.media.mit.edu/pia/Research/deepview/exif.html).

It's not fully fleshed out yet, but I wanted to check with you first, whether that scope extension is something you want to have in the library at all? 
If that's not the case, then I would need to see how I might re-use the work you already put into image detection to abstract it into it's own library, or we would eventually need to switch to our fork.

Let me know what you think,
Stephan

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ function asyncFileToBuffer (filepath, callback) {
         return callback(new Error('File size is not greater than 0 —— ' + filepath));
       }
       var bufferSize = Math.min(size, MaxBufferSize);
-      var buffer = new Buffer(bufferSize);
+      var buffer = new Buffer.alloc(bufferSize);
```
Comment (TimBeyer) [inline lib/index.js]:
```suggestion
      var buffer = Buffer.alloc(bufferSize);
```

File: lib/index.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ function syncFileToBuffer (filepath) {
   var descriptor = fs.openSync(filepath, 'r');
   var size = fs.fstatSync(descriptor).size;
   var bufferSize = Math.min(size, MaxBufferSize);
-  var buffer = new Buffer(bufferSize);
+  var buffer = new Buffer.alloc(bufferSize);
```
Comment (TimBeyer) [inline lib/index.js]:
```suggestion
  var buffer = Buffer.alloc(bufferSize);
```

File: lib/types/tiff.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ function readIFD (buffer, filepath, isBigEndian) {
   }
 
   // populate the buffer
-  var endBuffer = new Buffer(bufferSize);
+  var endBuffer = new Buffer.alloc(bufferSize);
```
Comment (TimBeyer) [inline lib/types/tiff.js]:
```suggestion
  var endBuffer = Buffer.alloc(bufferSize);
```

File: specs/fs-close.spec.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ var imageSize = require('..');
 
 describe('after done reading from files', function () {
   function readFromClosed (descriptor) {
-    fs.readSync(descriptor, new Buffer(1), 0, 1, 0);
+    fs.readSync(descriptor, new Buffer.alloc(1), 0, 1, 0);
```
Comment (TimBeyer) [inline specs/fs-close.spec.js]:
```suggestion
    fs.readSync(descriptor, Buffer.alloc(1), 0, 1, 0);
```

File: specs/others.spec.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ describe('Invalid invocation', function () {
       var bufferSize = 2048;
       var file = 'specs/images/valid/tiff/little-endian.tiff';
 
-      buffer = new Buffer(bufferSize);
+      buffer = new Buffer.alloc(bufferSize);
```
Comment (TimBeyer) [inline specs/others.spec.js]:
```suggestion
      buffer = Buffer.alloc(bufferSize);
```

File: specs/valid.spec.js [hunk 6]
```diff
@@ -105,7 +110,7 @@ describe('Valid images', function () {
 
       beforeEach(function (done) {
 
-        var buffer = new Buffer(bufferSize);
+        var buffer = new Buffer.alloc(bufferSize);
```
Comment (TimBeyer) [inline specs/valid.spec.js]:
```suggestion
        var buffer = Buffer.alloc(bufferSize);
```

**Comment by zcei** [comment]
@netroy any chance to get some feedback? Otherwise I would eventually need to fork, what would like to avoid if possible

**Comment by netroy** [comment]
@zcei sorry for the delay. I'll try to get this merged & released over the weekend.

**Comment by netroy** [comment]
@zcei would you mind, if I re-do this (I'll give you credit) ?
I'm not clear on how I want this code to look like, right now. But, I also don't want to merge this as it is.

PS: apologies for the delay on this. I'll work on getting this fixed as soon as I can.

**Comment by netroy** [comment]
merge in #141

**Comment by zcei** [comment]
Awesome, let me know what kind of changes you're interested in (ping me in an issue once you know) and I'll see whether I might can help cleaning this up 🙂

**Comment by netroy** [comment]
Honestly, the issue isn't just your changes. The entire code-base is still stuck in 2013.
I hope to change that soon.

Thanks for your contribution. I've published your fix in `0.7.1`.
Cheers

**Pull request closed without merging**